src/ir/block_builder_test.cc

```diff
@@ -0,0 +1,185 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#include "src/ir/block_builder.h"
+
+#include <memory>
+
+#include "src/common/testing/gtest.h"
+#include "src/ir/attribute.h"
+#include "src/ir/ir_context.h"
+#include "src/ir/types/entity_type.h"
+#include "src/ir/types/type.h"
+#include "src/ir/types/type_factory.h"
+
+namespace raksha::ir {
+namespace {
+
+class BlockBuilderTest : public testing::Test {
+ public:
+  BlockBuilderTest() {
+    context_.RegisterOperator(std::make_unique<Operator>("core.plus"));
+    context_.RegisterOperator(std::make_unique<Operator>("core.minus"));
+    context_.RegisterOperator(std::make_unique<Operator>("core.merge"));
+    context_.RegisterOperator(std::make_unique<Operator>("core.choose"));
```

Comment:
Probably not relevant to this PR, but seeing this usage is making me wonder if `Operator` should sent into `RegisterOperator` as a value and moved into the `unique_ptr` space via the move ctor. Would make it a little more concise to call `RegisterOperator`.

Reply:
I am not sure I quite follow. Are you saying an usage like this? `context_.RegisterOperator(Operator("core.plus"))`

Reply:
Yes, and then within `RegisterOperator`, we could have something like so:

```C++
void RegsiterOperator(Operator op) {
  auto insert_result = operator_map_.insert({op.name, std::make_unique<Operator>(std::move(op)));
  /*...*/
}
```

(The above may be slightly wrong, I didn't look at `RegisterOperator`'s decl or defn).